content/5_Plat/Bin_Jump.mdx

```diff
@@ -215,9 +214,100 @@ export const problems = {
 	<Resource source="SecondThread" title="Tree Basics - Binary Lifting" url="https://codeforces.com/blog/entry/81527"> </Resource>
 </Resources>
 
+<Info>
+
+Binary jumping is more commonly referred to as "binary lifting."
+
+</Info>
+
 ### Solution
 
-<IncompleteSection />
+To solve this problem, we can use a standard binary lifting implementation where ```query(int v, int j)``` corresponds to the $j$th ancestor of v.
+
+In our ```query(int v, int j)``` if we ever encounter a value of $-1$ or $0$, then such a node does not exist and we can simply return its value. This is because the lowest number a node can be is $1$ (the root of the tree).
+
+To calculate the columns required for the ```up``` array, use the formula $\textrm{ceil}(\log_2{N})+1$ which in our case simplifies to $\textrm{ceil}(\log_2{(2\cdot 10^5)}) + 1 = 19$.
+
+<Info>
+
+It never hurts to add additional columns (as long as it's reasonable)!
+
+</Info>
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define FOR(i, a, b) for(int i = (a); i < (b); i++)
+#define FORE(i, a, b) for(int i = (a); i <= (b); i++)
+#define F0R(i, a) for(int i = 0; i < (a); i++)
+#define trav(a, x) for (auto& a : x)
+
+int N, Q;
+
+const int MX = 2e5+5;
+
+int up[MX][20]; // 19 is calculated with our expression but we can add padding to be safe.
+
+vector<int> adj[MX]; // adjacency list
+
+void dfs(int v, int p) { // we keep track of the current node v and its parent p
+    up[v][0] = p;
+
+    FOR(i, 1, 20) {
+        if(up[v][i-1] != -1) up[v][i] = up[up[v][i-1]][i-1]; // construction of our ancestor tree
+        else up[v][i] = -1;
+    }
+
+    for(int e : adj[v]) if(e != p) dfs(e, v); // standard tree dfs
+}
+
+int query(int v, int j) { // a is our vertex, and j is the jump we make
+    if(v == -1 || j == 0) return v;
+    for(int i = 19; i >= 0; i--) { // start at our max value
+        if(j >= (1 << i)) return query(up[v][i], j - (1 << i));
+    }
+    return -1;
+}
+
+int main() {
+    ios_base::sync_with_stdio(0);
+    cin.tie(0);
+
+    cin >> N >> Q;
+    FOR(i, 2, N+1) { // one based indexing
+        int a; cin >> a;
+        adj[a].push_back(i);
+        adj[i].push_back(a);
+    }
```

Comment:
why not just set `up[i][0]` to `a` and evaluate the rest of the `up[i]` here?

Reply:
I just changed the implementation to be more consistent with sols but I suppose that would work too.